### ISSUE TITLE
Add documentation for CORS

### ISSUE THREAD

**Issue by sansyrox** [sparckles/Robyn#506]

**Comment by carlosm27** [comment]
I have a doubt about how to allow CORS in Robyn.

I tried this in the app.py:

```
app = Robyn(__file__)
ALLOW_CORS(app, origins=[])
app.add_request_header("Access-Control-Allow-Origin","*")

"Endpoints here"
```
And:
```
app = Robyn(__file__)
ALLOW_CORS(app, origins=["*"])
app.add_request_header("Access-Control-Allow-Origin","*")

"Endpoints here"
``
But when my React app make a request, it shows the error message about the CORS not allow.

The only way it works for me was:

```
@app.get("/book")
async def books():
    books = all_books()
    return Response(status_code = StatusCodes.HTTP_200_OK.value, headers= {"Access-Control-Allow-Origin":"*"}, body = books)
```

It is a way to allow CORS without explicitly allow it in every endpoint?

**Comment by sansyrox** [comment]
@carlosm27 , yes. You can do something like the following

```python
app = Robyn(__file__)
ALLOW_CORS(app, origins=["*"])
```

**Comment by carlosm27** [comment]
I tried that, but I receive the following message in the console:

![image](https://github.com/sansyrox/robyn/assets/70811425/66ff3389-8d4d-4e81-8540-f4af8210a62d)


This is my code:

```python
from robyn import Robyn, StatusCodes, ALLOW_CORS
from controllers import all_books, new_book, book_by_id, delete_book, update_book
import json
from robyn.robyn import Response

app = Robyn(__file__)


ALLOW_CORS(app, origins=["*"])


@app.post("/book")
async def create_book(request):
    
    body = request.body
    
    json_body = json.loads(body)
    
    try:
        book = new_book(json_body['title'], json_body['author'])
        return Response(status_code = StatusCodes.HTTP_200_OK.value, headers= {'Access-Control-Allow-Origin':"*"}, body = book)
    except:
        return Response(status_code = StatusCodes.HTTP_500_INTERNAL_SERVER_ERROR.value, headers= {"Access-Control-Allow-Origin":"*"}, body = "Internal Server Error")


@app.get("/book")
async def books(request):
    
    books = all_books()
    return Response(status_code = StatusCodes.HTTP_200_OK.value, headers= {"Access-Control-Allow-Origin":"*"}, body = books)


@app.delete("/book/:id")
async def delete(request):
    id = request.path_params["id"]

    book_id = book_by_id(id)

    if book_id == None:
        return Response(status_code = StatusCodes.HTTP_404_NOT_FOUND.value, headers= {"Access-Control-Allow-Origin":"*"}, body = "Book not Found")
    else:
        try: 
            delete_book(id)
            return Response(status_code = StatusCodes.HTTP_200_OK.value, headers= {"Access-Control-Allow-Origin":"*"}, body = "Book deleted")
        except:
            return Response(status_code = StatusCodes.HTTP_500_INTERNAL_SERVER_ERROR.value, headers= {"Access-Control-Allow-Origin":"*"}, body = "Internal Server Error")    



app.start(port=8000, url="0.0.0.0")

```

The issues I'm having, is that when the frontend make a GET request, there is not problem with CORS if I add `"Access-Control-Allow-Origin":"*"` to the headers attribute of the Response object. If I remove it, I have a CORS error message in the browser. But this alternative doesn't work with POST requests and DELETE requests.

**Comment by carlosm27** [comment]
> v0.47.0 should address the root of it. It should've been fixed even in v0.46.0 though. @carlosm27 , are you still facing this issue?

Hey @sansyrox, I still have CORS issues when using the DELETE and UPDATE methods. With GET and POST I don't have any CORS issues.
